docs/contracts.rst

```diff
@@ -990,12 +1007,17 @@ virtual method lookup.
 
 .. index:: ! constructor
 
+.. _constructor:
+
 Constructors
 ============
-A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation.
-Constructor functions can be either ``public`` or ``internal``. If there is no constructor, the contract will assume the
-default constructor: ``contructor() public {}``.
 
+A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation, and where you can run contract initialisation code. It always generates code that initialises any state variables you declare to their specified default value if you initialise them inline, or zero if you do not.
```

Comment:
In `It always generates` it is unclear what "it" refers to. Probably best to use passive form. Or start the sentence with "Other code that is executed once right before the constructor code executes is the initialisation of state variables..."